### PR TITLE
Remove support for !, || and &&

### LINKED ISSUE #1448: Remove !, || and &&
Remove support for `!`, `||` and `&&`. These should be replaced by the keywords `not`, `or`, and `and`.

This requires updates to documentation, website, and so forth too.

### PR DESCRIPTION
**Pull request by esbenbjerre**
WIP
Fixes #1448.

- [x] Remove support in parser
- [x] Remove usage in library
- [x] Remove usage in tests
- [ ] Update documentation
- [ ] Update website

### CODE REVIEW

**Comment by esbenbjerre** [comment]
I get some weird errors in a couple of tests now.
```
Core/Io/TestInputStream:
9676
-- Name Error -------------------------------------------------- main/test/ca/uwaterloo/flix/library/Core/Io/TestInputStream.flix
9677

9678
>> Undefined type variable 'a'.
9679

9680
140 |         g and a[0] == 0 as Int8 and a[1] == 1 as Int8 and a[2] == 2 as Int8 and a[3] == -1 as Int8
9681
                                          ^
9682
                                          undefined type variable.
9683
```
I'll try resetting my changes to the parser to verify if I messed something up.

**Comment by esbenbjerre** [comment]
I have concluded that the behaviour is as expected when using a keyword instead of the symbolic operator. I've added parentheses to expression in the failing test.

**Comment by magnus-madsen** [comment]
> 
> 
> I have concluded that the behaviour is as expected when using a keyword instead of the symbolic operator. I've added parentheses to expression in the failing test.

I guess it is a question of precedence? Did you see some tests, somewhere, with precedence?

**Comment by esbenbjerre** [comment]
> I guess it is a question of precedence? Did you see some tests, somewhere, with precedence?

No I did not. I just figured it was a non-issue since the expression seemed a bit ambiguous to me. But I think the thing messing it up is the combination with type cast. The program below parses, but replacing `&&` by `and` will not not parse.
```
def main(): Bool & Impure = {
    let a = [42 as Int8, 42 as Int8];
    a[0] == 42 as Int8 && a[1] == 42 as Int8
}
```

**Comment by magnus-madsen** [comment]
Is this PR ready for review?

**Comment by esbenbjerre** [comment]
> Is this PR ready for review?

It is, however I haven't updated doc etc. yet.

File: main/src/library/Int16.flix [hunk 1]
```diff
@@ -63,9 +63,9 @@ namespace Int16 {
     ///
     @Time(1) @Space(1)
     pub def dist(x: Int16, y: Int16): Int16 = {
-        if      (x >= 0i16 && y >= 0i16)                abs(x - y)
-        else if (x < 0i16 && y < 0i16)                  abs(x - y)
-        else if (x == minValue() || y == minValue())    -1i16
+        if      (x >= 0i16 and y >= 0i16)                abs(x - y)
+        else if (x < 0i16 and y < 0i16)                  abs(x - y)
+        else if (x == minValue() and y == minValue())    -1i16
```
Comment (magnus-madsen) [inline main/src/library/Int16.flix:68]: Mistake?

File: main/src/library/Int16.flix [hunk 2]
```diff
@@ -210,7 +210,7 @@ namespace Int16 {
     /// Returns 0 if `position` is outside the range 0-15 inclusive.
     ///
     def bitPositionToInt(position: Int32): Int16 =
-        if (position < 0 || position > size() - 1) 0i16 else 1i16 <<< position
+        if (position < 0 and position > size() - 1) 0i16 else 1i16 <<< position
```
Comment (magnus-madsen) [inline main/src/library/Int16.flix:213]: Mistake :)

File: main/src/library/Int16.flix (hunk 1 above)
Comment (esbenbjerre) [reply main/src/library/Int16.flix]: Woops.

File: main/src/library/Int16.flix (hunk 2 above)
Comment (esbenbjerre) [reply main/src/library/Int16.flix]: Woops 2.0.

**Comment by esbenbjerre** [comment]
I've taken another look at the changes and did not find further mistakes.

**Comment by magnus-madsen** [comment]
I am merging this now. Will you put as top priority to update the programming-flix book and the flix website?

**Comment by esbenbjerre** [comment]
> I am merging this now. Will you put as top priority to update the programming-flix book and the flix website?

Yep!

**Pull request merged**